### PR TITLE
initial SASS support

### PR DESCRIPTION
**Pull request by DevStarlight**
Added sass initial support to gulpfile in build/watch task
Added scss files in our initial pages
Added to gitignore package-lock.json

### CODE REVIEW

**Comment by DevStarlight** [comment]
Just in case you have problems updating your gulp build or gulp watch after updating npm.

I got problems by simply doing `npm install` as some post pointed out. I found the solution by removing the `node_modules` folder and also the `package-lock.json`. As well, I updated to the latest version by `npm install npm@latest`.

After that, the issue get resolved by itself by just doing `npm install`.

File: gulpfile.js [hunk 1]
```diff
@@ -72,23 +68,35 @@ var source_html = [
...
 
-
 gulp.task('default', ['watch']);
+
+var source_sass = [
+	'./pages/**/**.scss'
+];
+gulp.task('sass', function () {
+	gulp.src(source_sass)
+		.pipe(sass({outputStyle: 'expanded'}).on('error', sass.logError))
```
Comment (DevStarlight) [inline gulpfile.js:98]: here we have different options of the `outputStyle` that you can check and help deciding the way we produce & store our css files following [this](https://web-design-weekly.com/2014/06/15/different-sass-output-styles/) link.

**Comment by DevStarlight** [comment]
Further info about the gulp-sass plugin we use can be read by following this [link](https://github.com/dlmanning/gulp-sass#options)

**Comment by CodingCarlos** [comment]
**Nice job!** 👍 

To be fully functional, and merge it in dev, **it shall include styles into the HTML**. I would do that in the following way:

## Dev:
Instead of moving it to `/dist/css`, i would create a `/css` folder in the root, and move everyting there. That will make easy to add or remove stylesheets in development. All that stylesheets shall be added to index.html as 
```html
<link rel="stylesheet" type="text/css" href="css/pages/home.css" />
<link rel="stylesheet" type="text/css" href="css/pages/register.css" />
``` 

## Production
Create a new task piped to SASS who concat and minify all css files it can find in `./css`, and moves the result to `/dist/css/styles.css`, and add only one link in `/dist/index.html` file, pointing to that file. 

Maybe you have something different in mind. If so... just go for it ;)

**Comment by DevStarlight** [comment]
I've just upgraded this PR adding all what you said in your previous comment. Further info about the gulp-clean-css that we use to minify css can be found following [this](https://github.com/scniro/gulp-clean-css) link

**Comment by CodingCarlos** [comment]
Ready! I've added link also in `/dist/index.html`, so... It's done!!

**Pull request merged**